src/extractor/guidance/turn_handler.cpp

```diff
@@ -78,14 +78,32 @@ namespace guidance
 {
 
 // a wrapper to handle road indices of forks at intersections
-TurnHandler::Fork::Fork(const Intersection::iterator right, const Intersection::iterator left)
-    : right(right), left(left), size((left - right) + 1)
+TurnHandler::Fork::Fork(const Intersection::iterator intersection_base,
+                        const Intersection::iterator begin,
+                        const Intersection::iterator end)
+    : intersection_base(intersection_base), begin(begin), end(end), size(end - begin)
```

Comment:
should be fine, but I'd consider using `std::distance(begin,end)` here. Not really needed, but cleaner for different iterator types.